File: BiteClub/Models/OrderItem.swift
import Foundation

struct OrderItem: Codable, Hashable, Identifiable {
    var id: UUID
    var orderLocationId: UUID
    var userId: UUID
    var name: String
    var comments: String?
    var quantity: Int
    var createdAt: Date
    var updatedAt: Date
    
    var orderLocation: OrderLocation?
    var user: User?
    
    init(
        id: UUID?,
        orderLocationId: UUID,
        userId: UUID,
        name: String,
        comments: String? = nil,
        quantity: Int,
        createdAt: Date?,
        updatedAt: Date?,
        orderLocation: OrderLocation? = nil,
        user: User? = nil
    ) {
        self.id = id ?? UUID()
        self.orderLocationId = orderLocationId
        self.userId = userId
        self.name = name
        self.comments = comments
        self.quantity = quantity
        self.createdAt = createdAt ?? Date()
        self.updatedAt = updatedAt ?? Date()
        self.orderLocation = orderLocation
        self.user = user
    }
    
    
    enum CodingKeys: String, CodingKey {
        case id = "id"
        case orderLocationId = "order_location_id"
        case userId = "user_id"
        case name = "name"
        case comments = "comments"
        case quantity = "quantity"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case orderLocation = "order_location"
        case user = "user"
    }
}

File: BiteRunr/shared/src/commonMain/kotlin/org/biterunr_studios/biterunr/Models/OrderItem.kt
package org.biterunr_studios.biterunr.Models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.datetime.Instant

@Serializable
data class OrderItem(
    @SerialName("id")
    val id: String,

    @SerialName("order_location_id")
    val orderLocationId: String,

    @SerialName("user_id")
    val userId: String,

    @SerialName("name")
    val name: String,

    @SerialName("comments")
    val comments: String? = null,

    @SerialName("quantity")
    val quantity: Int,

    @SerialName("created_at")
    val createdAt: Instant,

    @SerialName("updated_at")
    val updatedAt: Instant,

    @SerialName("order_location")
    val orderLocation: OrderLocation? = null,

    @SerialName("user")
    val user: UserProfile? = null
)
